From 45ca0e6733d9b07ed87337c561b4ffc232ff087f Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Wed, 19 Feb 2020 09:56:24 +0200
Subject: [PATCH] [SW-8229] goya2: fix progress clearing mask

When clearing a mask of bits in progress, first shift mask
to correct place then invert mask to clear to zero.

Change-Id: I9466a35f93fd50f8aa02c707e4bb16ed305e2995
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 2f11680..11b92d5 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -40,7 +40,7 @@
 	uint32_t reg;
 
 	reg = regmap_readl(mmPSOC_SECURITY_SW_STATUS_0);
-	reg &= (~mask) << position;
+	reg &= ~(mask << position);
 	regmap_writel(reg, mmPSOC_SECURITY_SW_STATUS_0);
 }
 
